static/modes/glsl-mode.ts

```diff
@@ -0,0 +1,745 @@
+    addKeywords(vectorTypes('dvec'));
+    // GL_EXT_shader_explicit_arithmetic_types_float64
+    addKeywords(vectorTypes('f64vec'));
+    addKeywords(matrixTypes('f64mat'));
+    // GL_EXT_shader_explicit_arithmetic_types_float32
+    addKeywords(vectorTypes('f32vec'));
+    addKeywords(matrixTypes('f32mat'));
+    // GL_EXT_shader_explicit_arithmetic_types_float16
+    addKeywords(vectorTypes('f16vec'));
+    addKeywords(matrixTypes('f16mat'));
+    // GL_EXT_shader_explicit_arithmetic_types_int64
+    addKeywords(vectorTypes('i64vec'));
+    addKeywords(vectorTypes('u64vec'));
+    // GL_EXT_shader_explicit_arithmetic_types_int32
+    addKeywords(vectorTypes('i32vec'));
+    addKeywords(vectorTypes('u32vec'));
+    // GL_EXT_shader_explicit_arithmetic_types_int16
+    addKeywords(vectorTypes('i16vec'));
+    addKeywords(vectorTypes('u16vec'));
+    // GL_EXT_shader_explicit_arithmetic_types_int8
+    addKeywords(vectorTypes('i8vec'));
+    addKeywords(vectorTypes('u8vec'));
+
+    function samplerTypes(basename) {
+        return [`${basename}`, `${basename}Shadow`, `${basename}Array`, `${basename}ArrayShadow`];
+    }
+
+    addKeywords(samplerTypes('sampler1D'));
+    addKeywords(samplerTypes('isampler1D'));
+    addKeywords(samplerTypes('sampler2D'));
+    addKeywords(samplerTypes('samplerCube'));
+
+    function imageTypes(basename) {
+        return [
+            `${basename}1D`,
+            `${basename}1DArray`,
+            `${basename}2D`,
+            `${basename}2DArray`,
+            `${basename}2DRect`,
+            `${basename}2DMS`,
+            `${basename}2DMSArray`,
+            `${basename}3D`,
+            `${basename}Cube`,
+            `${basename}CubeArray`,
+            `${basename}Buffer`,
+        ];
+    }
+    addKeywords(imageTypes('image'));
+    addKeywords(imageTypes('iimage'));
+    addKeywords(imageTypes('uimage'));
+    addKeywords(imageTypes('texture'));
+    addKeywords(imageTypes('itexture'));
+    addKeywords(imageTypes('utexture'));
+    // GLSL_EXT_shader_image_int64
+    addKeywords(imageTypes('u64image'));
+    addKeywords(imageTypes('i64image'));
+
+    addKeywords([
+        'attribute',
+        'const',
+        'uniform',
+        'buffer',
+        'sampler',
+        'varying',
+        'lowp',
+        'mediump',
+        'highp',
+        'precision',
+        'invariant',
+        'discard',
+        'subroutine',
+        'layout',
+        'location',
+        'flat',
+        'centroid',
+        'texture',
+        'sample',
+        'patch',
+        'points',
+        'lines',
+        'triangles',
+        'max_vertices',
+        'max_primitives',
+        'local_size_x',
+        'local_size_y',
+        'local_size_z',
+        'local_size_x_id',
+        'local_size_y_id',
+        'local_size_z_id',
+        'precise',
+        'noperspective',
+        'writeonly',
+        'readonly',
+        'in',
+        'out',
+        'inout',
+
+        // scalar types
```

Comment:
Aren't a few of these already C++ keywords? Is it OK to put duplicates into monaco?

e.g.` const` and `float` at least

Reply:
yes, can remove some... I was just scanning up-and-down the GLSL list and tried to pull out the C/C++ based keywords

Reply:
I pulled out more (https://github.com/compiler-explorer/compiler-explorer/pull/6883/commits/9ebf47600640c034abb4781b9398ce1560d24bde), I guess not sure a better way to do a de-duplication check outside what I could grep from `monaco-editor/esm/vs/basic-languages/cpp/cpp`